chore(babel-preset-expo): remove babel decorators plugin (#27857)
expo/expo: packages/babel-preset-expo/README.md

Changes the engine preset in `@react-native/babel-preset` based on the JavaScript engine that is being targeted. In Expo SDK 50 and greater, this is automatically set based on the [`jsEngine`](https://docs.expo.dev/versions/latest/config/app/#jsengine) option in your `app.json`.

### `decoratorsPluginVersion`

> Changes version of proposal that `@babel/plugin-proposal-decorators` uses. Defaults to `legacy`. You can find more info about versions in [`@babel/plugin-proposal-decorators` docs](https://babeljs.io/docs/babel-plugin-proposal-decorators#version). It could be set to `false` to disable the plugin.

Reviewer: ```suggestion
Changes the proposal version that `@babel/plugin-proposal-decorators` uses. Defaults to `legacy`. Set to `false` to disable the plugin. Read more about supported versions in the [`@babel/plugin-proposal-decorators` docs](https://babeljs.io/docs/babel-plugin-proposal-decorators#version). 
```
